### PR TITLE
[DataPipe] Removing deprecated DataPipes

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71161
* #71171

We labeled these DataPipes as deprecated on Sep 30th, 2021 (#65827). Users should import these DataPipes from [TorchData](https://github.com/pytorch/data) to continue using them. We will be checking for any downstream library usage before landing this PR.

All deprecation related to DataPipes are tracked in https://github.com/pytorch/data/issues/163

DataPipes impacted by this PR:

| Name | Functional API  | Deprecation Date | What Happens Now 
| ------------- | ------------- | ------------- | ------------- |
| BucketBatcher | NA | Sep 30th, 2021 | Remove (moved to TorchData)
| HTTPReader | NA | Sep 30th, 2021 | Remove (moved to TorchData)
| LineReader | NA  | Sep 30th, 2021 | Remove (moved to TorchData)
| TarArchiveReader | NA | Sep 30th, 2021 | Remove (moved to TorchData)
| ZipArchiveReader | NA | Sep 30th, 2021 | Remove (moved to TorchData)

Lastly, tests related to those DataPipes will also be removed or migrated to TorchData (if they do not already exist there).

Differential Revision: [D33532272](https://our.internmc.facebook.com/intern/diff/D33532272)

cc @ezyang @gchanan @VitalyFedyunin @ejguan @NivekT

### CODE REVIEW

File: torch/utils/data/datapipes/iter/routeddecoder.py [hunk 1]
```diff
@@ -41,7 +40,6 @@ def __init__(self,
         if not handlers:
             handlers = (decoder_basichandlers, decoder_imagehandler('torch'))
         self.decoder = Decoder(*handlers, key_fn=key_fn)
-        deprecation_warning_torchdata(type(self).__name__)
```
Comment (NivekT) [inline torch/utils/data/datapipes/iter/routeddecoder.py:44]: @ejguan Is the plan to keep this DataPipe here in Core rather than TorchData?

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33532272).

**Comment by ejguan** [comment]
Do we have tracked how long the deprecation warning has been existing in PyTorch Core? We should wait at least one release cycle to remove them.

**Comment by NivekT** [comment]
> Do we have tracked how long the deprecation warning has been existing in PyTorch Core? We should wait at least one release cycle before removing them.

They were added on Sep 30, 2021 (close to 3.5 months). We can definitely wait a little bit longer if that is more appropriate.

**Comment by ejguan** [comment]
> > Do we have tracked how long the deprecation warning has been existing in PyTorch Core? We should wait at least one release cycle before removing them.
> 
> They were added on Sep 30, 2021 (close to 3.5 months). We can definitely wait a little bit longer if that is more appropriate.

If so, let's remove the DataPipe that contains deprecation warning in the release of 1.10. But, we should keep `routeddecoder` as the deprecation warning was added recently, it should be removed after 1.11, IMO.

**Comment by NivekT** [comment]
> > > Do we have tracked how long the deprecation warning has been existing in PyTorch Core? We should wait at least one release cycle before removing them.
> > 
> > 
> > They were added on Sep 30, 2021 (close to 3.5 months). We can definitely wait a little bit longer if that is more appropriate.
> 
> If so, let's remove the DataPipe that contains deprecation warning in the release of 1.10. But, we should keep `routeddecoder` as the deprecation warning was added recently, it should be removed after 1.11, IMO.
> 
> Edit: If we have lots of deprecation warnings in PyTorch core, we may want to create a table (google doc) to track the time for each deprecation warning. And then, remove the corresponding one before each future release. Don't spend time on it if this PR has already included all of these warnings.

I agree and will do.

**Comment by NivekT** [comment]
@ejguan Just to be clear, will we move `decoder` to TorchData, or will we remove it entirely from both repo? I am guessing it is the latter?

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33532272).

File: torch/utils/data/datapipes/iter/routeddecoder.py (hunk 1 above)
Comment (ejguan) [reply torch/utils/data/datapipes/iter/routeddecoder.py:44]: We should keep deprecation warning here.

**Review by ejguan — APPROVED** [review]
Thanks, LGTM with one comment below.

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33532272).

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33532272).

**Pull request closed without merging**